data/dataexchange/src/main/java/tech/pegasys/teku/data/SlashingProtectionImporter.java

```diff
@@ -0,0 +1,171 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.data;
+
+import static tech.pegasys.teku.data.slashinginterchange.Metadata.INTERCHANGE_VERSION;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.apache.logging.log4j.util.Strings;
+import org.apache.tuweni.bytes.Bytes;
+import tech.pegasys.teku.data.signingrecord.ValidatorSigningRecord;
+import tech.pegasys.teku.data.slashinginterchange.CompleteSigningHistory;
+import tech.pegasys.teku.data.slashinginterchange.InterchangeFormat;
+import tech.pegasys.teku.data.slashinginterchange.Metadata;
+import tech.pegasys.teku.data.slashinginterchange.MinimalSigningHistory;
+import tech.pegasys.teku.data.slashinginterchange.SignedAttestation;
+import tech.pegasys.teku.data.slashinginterchange.SignedBlock;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.logging.SubCommandLogger;
+import tech.pegasys.teku.provider.JsonProvider;
+
+public class SlashingProtectionImporter {
+  private final JsonProvider jsonProvider = new JsonProvider();
+  private Path slashingProtectionPath;
+  private List<MinimalSigningHistory> data = new ArrayList<>();
+  private Metadata metadata;
+  private final SubCommandLogger log;
+
+  public SlashingProtectionImporter(final SubCommandLogger log) {
+    this.log = log;
+  }
+
+  public void initialise(final File inputFile) throws IOException {
+    final ObjectMapper jsonMapper = jsonProvider.getObjectMapper();
+    try {
+      final JsonNode jsonNode = jsonMapper.readTree(inputFile);
+
+      metadata = jsonMapper.treeToValue(jsonNode.get("metadata"), Metadata.class);
+      if (metadata == null) {
+        log.exit(
+            1,
+            "Import file "
+                + inputFile.toString()
+                + " does not appear to have metadata information, and cannot be loaded.");
+        return; // Testing mocks log.exit
+      }
+      if (!metadata.interchangeFormatVersion.equals(INTERCHANGE_VERSION)) {
```

Comment:
```suggestion
      if (!INTERCHANGE_VERSION.equals(metadata.interchangeFormatVersion)) {
```

to avoid potential NullPointerException if the version is missing (Jackson might have caught it already but doesn't hurt to be paranoid)